Fixed #33026 -- Mentioned RequestFactory in testing tools docs. (#16094)
django/django: docs/topics/testing/tools.txt

* Use Django's test client to establish that the correct template is being
  rendered and that the template is passed the correct context data.

> * The :class:`~django.test.RequestFactory` is a trimmed down version of the
>   test client that bypasses the routing and middleware layers so you can test
>   your view as you would test any other function in python with known inputs, 
>   testing for specific outputs.

Reviewer: I think we could be a little more spare here. Maybe: 

```suggestion
* Use :class:`~django.test.RequestFactory` to test view functions directly, 
  bypassing the routing and middleware layers.  
```

Clicking the link you get the explanatory example in the [Advanced testing tools page](https://docs.djangoproject.com/en/4.1/topics/testing/advanced/#the-request-factory).

* Use Django's test client to establish that the correct template is being
  rendered and that the template is passed the correct context data.

* The :class:`~django.test.RequestFactory` is a trimmed down version of the test client that bypasses the routing and middleware layers
  so you can test your view as you would test any other function in python with known inputs, 

>   testing for specific outputs. More detail on this can be found in :doc:`advanced testing topics</topics/testing/advanced>`.

Reviewer 2: Wrap at 79 chars.

Reviewer 2: I'm not sure if the last sentence adds value, as `` :class:`~django.test.RequestFactory` `` refers to exactly the same page :thinking: .

* Use in-browser frameworks like Selenium_ to test *rendered* HTML and the
  *behavior* of web pages, namely JavaScript functionality. Django also
  provides special support for those frameworks; see the section on
  :class:`~django.test.LiveServerTestCase` for more details.

> A comprehensive test suite should use a combination of all of these test types.

Reviewer 2: Chop unnecessary blank lines and wrap docs at 79 chars.
